app/controllers/challenges_controller.rb

```diff
@@ -19,7 +19,8 @@ def index
 
   # GET
   def entries
-    @challengeEntries = ChallengeEntry.includes(:submission).where(challenge_id: @challenge.id).order('created_at DESC').paginate(page: params[:page], per_page: 25).includes(submission: :comments)
+    #@challengeEntries = ChallengeEntry.includes(:submission).where(challenge_id: @challenge.id).order('created_at DESC').paginate(page: params[:page], per_page: 25).includes(submission: :comments)
+    @entries = base_submissions.includes(:challenge_entries).where("challenge_entries.challenge_id = #{@challenge.id}").order('challenge_entries.created_at DESC').paginate(page: params[:page], per_page: 25)
```

Comment:
This is a clever way of doing this, but I'd want to make sure it does what you *think* it does.
In my experience, Rails turns `.includes` into a *second* database query, so first it would query *all* submissions, then query the challenge entries for *all* submissions, *then* filter. You're going to want to make sure that's not what this does.
That said, I don't think this inverted way of doing things is the best way either way: you probably want to use Challenge `has_many :submissions, through: :challenge_entries` (and then `@challenge.includes(challenge_entries: :submission`), or `ChallengeEntries.where(challenge: @challenge).order('created_at DESC').includes(:submission)` then `map(&:submission)` (that's probably what I'd do, unless it's possible to set an order through a model association, which isn't something I've tested).
Though, if the way you're using here works the way you'd want it to (and not the messed up way I described), then feel free to disregard this.